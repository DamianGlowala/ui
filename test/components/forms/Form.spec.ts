import { describe, it, expect } from 'vitest'
import { UForm } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Form', () => {
  it.each([
    ['basic case', {}],
    ['with default slot', { slots: { default: 'Form slot' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Form.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UForm)
    expect(html).toMatchSnapshot()
  })
})
