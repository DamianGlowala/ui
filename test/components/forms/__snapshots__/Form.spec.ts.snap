// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form > renders basic case correctly 1`] = `"<form></form>"`;

exports[`Form > renders with default slot correctly 1`] = `"<form>Default slot</form>"`;
